demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -287,28 +289,41 @@ def check_dataset_exists(
                     query, print_req_error=(i + 1 == timeout // interval)
                 )
                 results = xsiam_client.get_xql_query_result(execution_id)
+                # if we got result we will break from the loop
                 if results:
                     logger.info(
                         f"[green]Dataset {dataset} exists[/green]",
                         extra={"markup": True},
                     )
                     dataset_exist = True
+                    results_exist = True
+                    break
+                # if we don't have results from the dataset immediately we will continue to try until the timeout.
+                # if we don't have any results until the timeout dataset_exist is set to False and we will raise an error.
                 else:
-                    err = (
-                        f"[red]Dataset {dataset} exists but no results were returned. This could mean that your testdata "
-                        "does not meet the criteria for an associated Parsing Rule and is therefore being dropped from "
-                        "the dataset. Check to see if a Parsing Rule exists for your dataset and that your testdata "
-                        "meets the criteria for that rule.[/red]"
+                    dataset_exist = True
+                    results_exist = False
+                    logger.info(
+                        f"[cyan]try to get results from the data set, there are not results for the {i+1}th time. continue.[/cyan]",
+                        extra={"markup": True},
                     )
-                    logger.error(err, extra={"markup": True})
-                break
             except requests.exceptions.HTTPError:
                 pass
             sleep(interval)
+        # There are no results from the dataset but it exists. If the dataset doesn't exist HTTPError exception is raised.
```

Comment:
The dataset exists here, but no results. The second part of the comment here is wrong.